app/models/vm_or_template.rb

```diff
@@ -1923,6 +1923,10 @@ def validate_supported
     {:available => true,   :message => nil}
   end
 
+  def validate_unsupported(message_prefix)
```

Comment:
Ah, I see. So, it really should be moved from `vm/operations.rb` to `vm_or_template/operations.rb`.